.github/workflows/pythonpackage.yml

```diff
@@ -34,5 +34,5 @@ jobs:
       run: |
         pip install -e .
         pip install compressed_rtf
-        pip install pytest pytest-console-scripts
+        pip install pytest mock==3.0.5 pytest-console-scripts
```

Comment:
What if we have a separate line here that first installs mock==3.0.5 and only after that another line that installs pytest & ... console-scripts ? Would that work?

Reply:
Does not seem to work, either.

I just started with gh actions recently after reading @hynek s article
https://hynek.me/articles/python-github-actions/

Ok, I guess the problem is not the custom installations, but already the `pip install -e` - it crashes directly after the installation of `pytest-runner` - which I am not familiar with.
see
https://github.com/jugmac00/tnefparse/runs/689783960?check_suite_focus=true

At `pytest-runner`s repo you can read its usage is deprecated.

I am still not sure why it works on my machine, but fails on gh action.

I try to have another look in the next couple of days.

Reply:
`'mock>=2.0.0'` is coming from `pytest-console-scripts` - another plugin I am not familiar with.

```
Installed /home/runner/work/tnefparse/tnefparse/.eggs/pytest_console_scripts-0.2.0-py2.7.egg
```

But version 0.2.0 of pytest-console-script is Python 3 only (when I look at the classifiers), and this requires only `mock>=2.0.0`. I guess the problem is not your script, but one of your dependencies.

Unfortunately, I have not more time atm.

Reply:
Hm, you put `pytest-runner` and `pytest-console-scripts` in your `setup_requires` - which I never used before.

Having a look at
https://pip.pypa.io/en/stable/reference/pip_install/#controlling-setup-requires

> Setuptools offers the setup_requires setup() keyword for specifying dependencies that need to be present in order for the setup.py script to run. Internally, Setuptools uses easy_install to fulfill these dependencies.

I do some Zope development, and we use buildout there - and buildout does not use pip, but `setuptools.easy_install` - and we suffer also from the problem, that for our Python 2.7 branches Python 3 packages get installed (unless we pin them in buildout.cfg).

So maybe the solution is not to use `setup_requires` - that is something I have to read more about.

But this looks like a good trace.